### PR TITLE
Update softhsm and libp11 to latest

### PR DESCRIPTION
**Pull request by ricardosalveti**
- softhsm: bump to 2.5.0
- libp11: bump to git 57ca68f for EC signing support

This also enables support for using EdDSA, which is available in latest OpenSSL (1.1.1a).

### CODE REVIEW

**Comment by patrickvacek** [comment]
Cool! It was about time to get some of these up to date. Especially cool to be able to get rid of the patches and to see EdDSA support! I'm running tests now.

File: recipes-support/softhsm/softhsm_git.bb [hunk 1]
```diff
@@ -1,27 +1,22 @@
...
-EXTRA_OECONF = "--disable-gost --with-openssl=${STAGING_LIBDIR}/.."
+# EdDSA requires OpenSSL >= 1.1.1
+EXTRA_OECONF = "--enable-eddsa --disable-gost"
 
-do_configure() {
- unset docdir
- sh ./autogen.sh
- oe_runconf
+do_configure_prepend() {
+    (cd ${S}; unset docdir; sh ./autogen.sh; cd -)
```
[the remark this answers is not in the document]
Comment (ricardosalveti) [reply recipes-support/softhsm/softhsm_git.bb]: True, let me update, thanks!

**Comment by lbonn** [comment]
Yes, that's good!

My only concern is whether it would make backports to rocko too difficult (we try to have minimal differences in our recipes, in particular with relation to dependencies versions). In particular, dropping the OpenSSL 1.0.2 patch and the `--enable-eddsa` configuration flag.

Maybe @OYTIS has a clearer picture about that?

**Comment by ricardosalveti** [comment]
> My only concern is whether it would make backports to rocko too difficult (we try to have minimal differences in our recipes, in particular with relation to dependencies versions). In particular, dropping the OpenSSL 1.0.2 patch and the --enable-eddsa configuration flag.

That is why I added that comment in there, to avoid issues when backporting to other branches, but it would still be something to change when doing the actual backport.

Ideally softhsm should be able to handle this automatically, but that still requires https://github.com/opendnssec/SoftHSMv2/pull/438 to be merged (didn't test if that works well when cross building).

**Comment by patrickvacek** [comment]
Passed oe-selftest. Still requires some manual testing with the HSM functionality, though.

I don't think the backport issues are big enough to prevent merging, but we should discuss and decide how to handle things like this.

**Pull request merged**